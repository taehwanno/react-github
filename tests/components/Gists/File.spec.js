import React from 'react';

import ReactGithub, { File, Gist } from '../../../src';

describe('<File />', () => {
  it('should throw an error when rendered alone', () => {
    expect(() => {
      ReactGithub.render(<File />);
    }).toThrowErrorMatchingSnapshot();
  });

  it('should throw an error when props.name not exist or not string', () => {
    expect(() => {
      ReactGithub.render((
        <Gist>
          <File />
        </Gist>
      ));
    }).toThrowErrorMatchingSnapshot();
  });

  it('should throw an error when props.children not exist or not string', () => {
    expect(() => {
      ReactGithub.render((
        <Gist>
          <File name="file.txt" />
        </Gist>
      ));
    }).toThrowErrorMatchingSnapshot();
  });

  it('should throw an error when the same props.name exist among <File />', () => {
    expect(() => {
      ReactGithub.render((
        <Gist>
          <File name="file.txt">contents</File>
          <File name="file.txt">contents</File>
        </Gist>
      ));
    }).toThrowErrorMatchingSnapshot();
  });

  it('should match snapshot with props.name and props.children', () => {
    expect(ReactGithub.render((
      <Gist>
        <File name="file1.txt">contents of file1.txt</File>
      </Gist>
    ))).toMatchSnapshot();
  });
});
