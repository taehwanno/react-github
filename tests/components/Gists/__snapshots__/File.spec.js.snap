// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<File /> should match snapshot with props.name and props.children 1`] = `
Object {
  "description": "",
  "files": Object {
    "file1.txt": Object {
      "content": "contents of file1.txt",
    },
  },
  "public": true,
}
`;

exports[`<File /> should throw an error when props.children not exist or not string 1`] = `"props.children in <File /> must be required as string"`;

exports[`<File /> should throw an error when props.name not exist or not string 1`] = `"props.name in <File /> must be required as string"`;

exports[`<File /> should throw an error when rendered alone 1`] = `"<File /> can be only rendered within <Gist />"`;

exports[`<File /> should throw an error when the same props.name exist among <File /> 1`] = `"the same props.name is not allowed among <File />"`;
